### ISSUE TITLE
Replace JSR-305 for checkerframework

### ISSUE THREAD

**Issue by gvdutra** [sirixdb/sirix#467]
### Feature idea

http://findbugs.sourceforge.net/index.html has not been maintained anymore it stops on java 8. People are now using spotbugs which is a fork and continuation of findbugs. https://spotbugs.github.io/

The idea here is to replace findbugs artifactory for spotbugs.

**Comment by gvdutra** [comment]
@JohannesLichtenberger can you please assign it to me? Thank you.

**Comment by JohannesLichtenberger** [comment]
Sure, thanks a lot for your contributions :+1:

**Comment by gvdutra** [comment]
Yes! I will do it, sorry I am a bit busy.... :)

**Comment by JohannesLichtenberger** [comment]
No pressure, just wanted to ask ;)

**Comment by gvdutra** [comment]
@JohannesLichtenberger sirix project is using JSR-305 for the annotations, not the "entire" findbugs, so the spotbugs is not ideal as still uses JSR-305.

JSR-305 is in dormant status even at spotbugs it says that to keep using JSR-305 from findbugs. , so that I was searching for a better solution, and I found it! there is a project called https://checkerframework.org/manual/ that is sort of slf4j for logging framework, whereas the "logging framework" here is all those validation in form of annotation. So that, I decided to move to checkerframework, I think this is the better solution IMO. 

Lemme know what you think, and sorry for the delay, I was trying to find a better solution for it.
